code/modules/mob/living/carbon/human/ai/brain/ai_brain.dm

```diff
@@ -99,41 +119,40 @@ GLOBAL_LIST_EMPTY(human_ai_brains)
 	nade_throwback(things_around)
 	item_search(things_around)
 
-	if(!currently_busy && should_reload_primary())
-		currently_busy = TRUE
+	if(!currently_reloading && should_reload_primary())
```

Comment:
with this, doesn't it mean that they could be busy with some other action and then decide to try and reload?

Reply:
Indeed, an error of mine